### ISSUE TITLE
Set return types of `fromJsonFloatString` and `fromJsonDecimalString` as `JsonFloat` and `JsonDecimal` respectively

### ISSUE THREAD

**Issue by chiranSachintha** [ballerina-platform/ballerina-lang#25446]
**Description:**
```ballerina
public type JsonDecimal ()|boolean|string|decimal|JsonDecimal[]|map<JsonDecimal>;
public function fromJsonDecimalString(string str) returns JsonDecimal|error = external;

public type JsonFloat ()|boolean|string|float|JsonFloat[]|map<JsonFloat>;
public function fromJsonFloatString(string str) returns JsonFloat|error = external;
```
To fix this issue need to allow cyclic reference in unions typedefs.

**Comment by KRVPerera** [comment]
This is getting fixed by Issue
https://github.com/ballerina-platform/ballerina-lang/issues/27899

**Comment by MaryamZi** [comment]
Fixed in #28714, merged with #29119.
